### ISSUE TITLE
Multisite broken on windows

### ISSUE THREAD

**Issue by rrelmy** [statamic/cms#1933]
## Bug Description
When running a multisite on windows the frontend only responds with 404 on every page.
Also the CP only shows empty collections.

The same site works on mac and linux.

## How to Reproduce

```
composer create-project statamic/statamic multi-bug-repro --prefer-dist --stability=dev
cd multi-bug-repro
php please multisite
// edit config as described
php artisan serve
```

Visit `http://localhost:8000/`

## Extra Detail
The stache seems to be empty, clearing config, cache or stache does not help.

I am willing to dig into the problem but I need some guidance where to start.

## Environment

**Statamic version:** 3.0.0.-beta-31

**PHP version:** 7.4.7

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by jasonvarga** [comment]
You may have done nothing wrong, and I haven't tried it out yet, but - what does your sites.php config file look like?

**Comment by rrelmy** [comment]
The `sites.php` looks like 

``` php
<?php

return [
    'sites' => [
        'default' => [
            'name' => config('app.name'),
            'locale' => 'en_US',
            'url' => '/',
        ],
        'two' => [
            'name' => 'two',
            'locale' => 'en_US',
            'url' => '/two/',
        ],
    ]
];
```

**Comment by rrelmy** [comment]
I believe I found the underlying problem the string comparsion inside https://github.com/statamic/cms/blob/389b90cbc8fea2c72596e12042b58dd7b4119bc2/src/Stache/Stores/CollectionEntriesStore.php#L30 retrieves some paths with slashes some with backslashes.
The comparsion fails and nothing is loaded.

This very naive workaround seems to fix the issue
``` php
$relative = str_replace('\\', '/', $file->getPathname());
```

After making this change sites show up.
I did not test anything else yet.
